File: P8_0402/P8_0402.cpp
#include<stdio.h>
int main()
{
    int yr;
    scanf("%d",&yr);
    int flag=0;
    
    while (flag==0)
    {
        yr++;
        int unit = yr%10;
        int ten  = (yr%100)/10;
        int hund = (yr%1000)/100;
        int thous= yr/1000;
        if (unit==ten || ten==thous || thous==hund || unit==thous|| unit==hund|| ten==hund)
        {
            flag=0;
        }
        else{
            flag=1;
        }
            
    }
    printf("Unique year: %d",yr);   
}

File: P8_0402/P8_0402.py
yr = int(input())
flag=0
while(flag==0):
    yr+=1
    unit = yr%10;
    ten  = (yr%100)//10;
    hund = (yr%1000)//100;
    thous= yr//1000;    
    if(unit==ten or ten==thous or thous==hund or unit==thous or unit==hund or ten==hund):
	flag=0
    else:
	flag=1
print("Unique Year:",yr)